pkg/services/ngalert/remote/client/alertmanager_configuration.go

```diff
@@ -39,13 +39,13 @@ func (mc *Mimir) GetGrafanaAlertmanagerConfig(ctx context.Context) (*UserGrafana
 	return gc, nil
 }
 
-func (mc *Mimir) CreateGrafanaAlertmanagerConfig(ctx context.Context, c, hash string, id, created int64, d bool) error {
+func (mc *Mimir) CreateGrafanaAlertmanagerConfig(ctx context.Context, cfg, hash string, id, createdAt int64, isDefault bool) error {
```

Comment:
Just some variable name changes. I updated the variable names in the `MimirClient` interface to be a bit more descriptive, I did the same for this method.

Reply:
Single-letter variable names are pretty common in Go. In particular on methods that have the type right next to them and get assigned to something else (e.g. the `UserGrafanaConfig`) right after.

Personally, I'm not a big fan of this and much prefer the single variable approach. You can read https://google.github.io/styleguide/go/decisions.html#single-letter-variable-names

I'm not fussed and you're welcome to change it - but I think this is the perfect use-case for that.

Reply:
I agree in cases such as `r *http.Request`, `for i, v := range values`, or for method receivers, but here it's not obvious what each variable means. From the style guide:
> Limit their use to instances where the full word is obvious and where it would be repetitive for it to appear in place of the single-letter variable.

If for example `c` was of type `*Config` this would be a different story, but since the arguments are primitives there's nothing suggesting that `c` stands for "configuration" or `d` for "default". I had to manually check what each value meant in the struct that implemented this interface and I want to save that cognitive load to people reading this code in the future.